### ISSUE TITLE
Unclear effects of color, fill and width parameters when drawing borderless shapes

### ISSUE THREAD

**Issue by Rodrigodd** [pymupdf/PyMuPDF#3290]
### Description of the bug

If you explicitly pass `color=None` to `Shape.finish()`, after a `draw_polyline` (or any other drawing call, I believe), the value passed in `width` is ignored, and a black line with `width=1.0` is drawn instead.

The [docs for `Shape.finish`](https://pymupdf.readthedocs.io/en/latest/shape.html#Shape.finish) say that the default value of `color` is `None`, but looking at the source code, the default value is actually `(0,)`. Also, I could not find documentation for the parameter `width`.

### How to reproduce the bug

Run the following code:

```python
import fitz
from pprint import pprint

pdf = fitz.open()
page = pdf.new_page(width=512, height=512)
shape: fitz.Shape = page.new_shape()  # make a drawing canvas for the output page

shape.draw_polyline([(64, 64), (192, 64), (192, 192), (64, 192), (64, 64)])

shape.finish(width=0.1)

shape.draw_polyline([(256, 64), (384, 64), (384, 192), (256, 192), (256, 64)])

shape.finish(color=None, width=0.1)

shape.commit()
pdf.save("Frot.pdf")
```

This will create a PDF that renders the following:

![image](https://github.com/pymupdf/PyMuPDF/assets/51273772/d17eac38-83d5-415b-96e6-065829746500)

### PyMuPDF version

1.23.25

### Operating system

Windows

### Python version

3.12

**Comment by JorjMcKie** [comment]
The documentation is indeed incomplete / outdated:
1. A borderless shape is drawn if and only if `fill` is not `None` and width is zero. In this case, the value of parameter `color` is ignored.
3. The default for `color` indeed is `(0,)` - black. Used to be `None` and will be updated in one of the next documentation update round.

**Comment by JorjMcKie** [comment]
Fixed in v1.24.2.
